File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        if not prices:
            return 0

        lowest_price = prices[0]
        max_profit = 0

        for i in range (len(prices)):
            lowest_price = min(lowest_price, prices[i])
            max_profit = max(max_profit, prices[i] - lowest_price)

        return max_profit    






        

File: Data Structures & Algorithms/maximum-difference-between-even-and-odd-frequency-i/submission-0.cpp
#include<map>
#include<string>
#include<algorithm>
using namespace std;
class Solution {
public:
    int maxDifference(string s) {
        map<char,int> freq;
        for(char c: s){
            freq[c]++;
        }
        int maxOdd = 0;
        int minEven = 101;

        for(const auto& pair : freq){
            if(pair.second %2 == 0){
                minEven = min(minEven,pair.second);
            }
            else{
                maxOdd = max(maxOdd, pair.second);
            }
        }
        return maxOdd-minEven;
    }
};